File: clojure/day01/test/day01/solutions_test.clj
(ns day01.solutions-test
  (:require [clojure.test :refer :all]
            [day01.solutions :refer :all]))

(deftest test-01-01
  (testing "Test 01 - 01"
    (is (= (solution1 "(())") 0))))

(deftest test-01-02
  (testing "Test 01 - 02"
    (is (= (solution1 "(((") 3))))

(deftest test-01-03
  (testing "Test 01 - 03"
    (is (= (solution1 "()()") 0))))

(deftest test-01-04
  (testing "Test 01 - 04"
    (is (= (solution1 ")))(((") 0))))

(deftest test-01-05
  (testing "Test 01 - 05"
    (is (= (solution1 ")))((((((") 3))))

(deftest test-01-06
  (testing "Test 01 - 06"
    (is (= (solution1 "((()))(((") 3))))

(deftest test-02-01
  (testing "Test 02 - 01"
    (is (= (solution2 "()())") 5))))

(deftest test-02-02
  (testing "Test 02 - 02"
    (is (= (solution2 ")") 1))))

(deftest test-02-03
  (testing "Test 02 - 03"
    (is (= (solution2 "(((())))))") 9))))

(deftest test-02-04
  (testing "Test 02 - 04"
    (is (= (solution2 "(())()))))") 7))))

(deftest test-02-05
  (testing "Test 02 - 05"
    (is (= (solution2 "(())(())(()))(((") 13))))

(deftest test-02-06
  (testing "Test 02 - 06"
    (is (= (solution2 "())") 3))))

File: java11/day01/src/test/java/day01/SolutionsTest.java
package day01;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class SolutionsTest {

  @Test
  public void testSolution1_1() {
    assertEquals(0, Solutions.solution1("(())"));
    assertEquals(0, Solutions.solution1("()()"));
  }

  @Test
  public void testSolution1_2() {
    assertEquals(3, Solutions.solution1("((("));
    assertEquals(3, Solutions.solution1("(()(()("));
    assertEquals(3, Solutions.solution1("))((((("));
  }

  @Test
  public void testSolution1_3() {
    assertEquals(-1, Solutions.solution1("())"));
    assertEquals(-1, Solutions.solution1("))("));
  }

  @Test
  public void testSolution1_4() {
    assertEquals(-3, Solutions.solution1(")))"));
    assertEquals(-3, Solutions.solution1(")())())"));
  }

  @Test
  public void testSolution2_1() {
    assertEquals(1, Solutions.solution2(")"));
  }

  @Test
  public void testSolution2_2() {
    assertEquals(5, Solutions.solution2("()())"));
  }

  @Test
  public void testSolution2_3() {
    assertEquals(9, Solutions.solution2("(((())))))"));
  }

  @Test
  public void testSolution2_4() {
    assertEquals(7, Solutions.solution2("(())()))))"));
  }

  @Test
  public void testSolution2_5() {
    assertEquals(13, Solutions.solution2("(())(())(()))((("));
  }
}
